spezi-module/onboarding/src/main/java/edu/stanford/spezikt/spezi_module/onboarding/invitation/InvitationCodeViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package edu.stanford.spezikt.spezi_module.onboarding.invitation
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import edu.stanford.spezikt.coroutines.di.Dispatching
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class InvitationCodeViewModel @Inject internal constructor(
+    private val iam: InvitationAuthManager,
+    @Dispatching.IO private val scope: CoroutineScope,
```

Comment:
there is no need to introduce a coroutine scope in view models. In view models we should use `viewModelScope` (via `implementation(libs.androidx.lifecycle.view.model.ktx)`) which is bound to the lifecycle of the view model, and cancels any pending actions when the view model gets cleared